pkg/rule/repo.go

```diff
@@ -171,20 +175,38 @@ func (r Rule) Run(input Parameters, ctx Context, decisions []int) (Parameters, [
 		}
 	}
 
-	params := Parameters{}
+	var (
+		params = Parameters{}
+		d      = decisions
+	)
+
+	diceRollout := randInt()
+	if len(decisions) != 0 {
+		diceRollout = decisions[0]
+	}
 
 	switch r.kind {
 	case KindOverride:
 		params = r.buckets[0].Parameters
 	case KindExperiment:
 		return Parameters{}, nil, errors.New("experiment based rules not implemented")
 	case KindRollout:
-		return Parameters{}, nil, errors.New("rollout based rules not implemented")
+		if len(decisions) != 0 {
+			d = decisions
+		} else {
+			d = append(d, diceRollout)
+		}
+
+		if diceRollout <= int(r.rollout) {
+			params = r.buckets[0].Parameters
+		} else {
+			return nil, nil, errors.Wrap(errors.ErrRuleNotInRollout, "rollout percentage")
```

Comment:
Even so not included in the rollout we need to return the decision that was made so we can store it and don't re-run it on next render.